blueprints/minimal/templates/config.cue

```diff
@@ -0,0 +1,104 @@
+package templates
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	timoniv1 "timoni.sh/core/v1alpha1"
+)
+
+// Config defines the schema and defaults for the Instance values.
+#Config: {
+	// The kubeVersion is a required field, set at apply-time
+	// via timoni.cue by querying the user's Kubernetes API.
+	kubeVersion!: string
+	// Using the kubeVersion you can enforce a minimum Kubernetes minor version.
+	// By default, the minimum Kubernetes version is set to 1.20.
+	clusterVersion: timoniv1.#SemVer & {#Version: kubeVersion, #Minimum: "1.20.0"}
+
+	// The moduleVersion is set from the user-supplied module version.
+	// This field is used for the `app.kubernetes.io/version` label.
+	moduleVersion!: string
+
+	// The Kubernetes metadata common to all resources.
+	// The `metadata.name` and `metadata.namespace` fields are
+	// set from the user-supplied instance name and namespace.
+	metadata: timoniv1.#Metadata & {#Version: moduleVersion}
+
+	// The labels allows adding `metadata.labels` to all resources.
+	// The `app.kubernetes.io/name` and `app.kubernetes.io/version` labels
+	// are automatically generated and can't be overwritten.
+	metadata: labels: timoniv1.#Labels
+
+	// The annotations allows adding `metadata.annotations` to all resources.
+	metadata: annotations?: timoniv1.#Annotations
+
+	// The selector allows adding label selectors to Deployments and Services.
+	// The `app.kubernetes.io/name` label selector is automatically generated
+	// from the instance name and can't be overwritten.
+	selector: timoniv1.#Selector & {#Name: metadata.name}
+
+	// The image allows setting the container image repository,
+	// tag, digest and pull policy.
+	// The default image repository and tag is set in `values.cue`.
+	image!: timoniv1.#Image
+
+	// The pod allows setting the Kubernetes Pod annotations, image pull secrets,
+	// affinity and anti-affinity rules. By default, pods are scheduled on Linux nodes.
+	pod: {
+		annotations?: timoniv1.#Annotations
+
+		affinity: *{
+			nodeAffinity: requiredDuringSchedulingIgnoredDuringExecution: nodeSelectorTerms: [{
+				matchExpressions: [{
+					key:      corev1.#LabelOSStable
+					operator: "In"
+					values: ["linux"]
+				}]
+			}]
+		} | corev1.#Affinity
+
+		imagePullSecrets?: [...timoniv1.#ObjectReference]
+	}
+
+	// The resources allows setting the container resource requirements.
+	// By default, the container requests 10m CPU and 32Mi memory.
+	resources: timoniv1.#ResourceRequirements & {
```

Comment:
This comment applies to lines 45 and 65 as well. 

I've been going back and forth on where these types of things should be configured. In a lot of cases, you can make very good guess about things like security contexts and resource constraints. Defaulting these reduces the cognitive load on a consumer to figure these out.

However, should these be defined in `config.cue` or `values.cue`? The `values.cue` technically contains the "defaults" according to the documentation. So, as an end-user, I'm more likely to look there than in `config.cue`. As it stands, we're sort of intermixing defaults. The `config.cue` holds both the schema and some default configuration which then has the defaults of `values.cue` applied over it. 

Hopefully, that makes sense. All that to say, I'm leaning towards moving these to `values.cue` and keeping `config.cue` purely schema driven.

Reply:
I suppose this would also apply to things like the default service port as well. I guess the question becomes where do you add a default value: `config.cue` or `values.cue`?. If they are literally interchangeable (and perhaps there's a distinction I'm not understanding here), then it seems like without standardizing on one you're going to end up looking in two places to understand a module's default values.

Reply:
The `values.cue` serves as a default file when no values files are supplied by the user, and the fields there get overwritten not unified. The `config.cue` is the place where CUE defaults can be set. The reason I've put the image concrete values in `values.cue` is for Dependabot or other automation to bump them without having to parse CUE, just JSON. I get that this is very confusing and I haven't got to the Dependabot part yet. I will change the blueprint to set the image defaults in `config.cue`.

Reply:
> I'm leaning towards moving these to `values.cue` and keeping `config.cue` purely schema driven.

I suggest you use a dedicated file for this, like `defaults.cue` and leave `values.cue` empty. As documented in the header, `values.cue` can't used for anything but concrete fields.

```
// Note that this file must have no imports and all values must be concrete.
```